File: Android/src/main/cs/DiscordRPCService.cs
using System;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using AndroidX.Core.App;

// фоновый сервис для работы с discord rpc на c#
namespace StormDiscordRPC
{
    [Service]
    public class DiscordRPCService : Service
    {
        private const string CHANNEL_ID = "DiscordRPCChannel";
        private const int NOTIFICATION_ID = 1;
        
        private DiscordRPCManager discordRPCManager;
        private bool isRunning = false;

        public override void OnCreate()
        {
            base.OnCreate();
            CreateNotificationChannel();
            discordRPCManager = new DiscordRPCManager();
        }

        public override StartCommandResult OnStartCommand(Intent intent, StartCommandFlags flags, int startId)
        {
            if (intent != null && !isRunning)
            {
                // получаем данные из интента
                string status = intent.GetStringExtra("status");
                string details = intent.GetStringExtra("details");
                string state = intent.GetStringExtra("state");

                // запускаем сервис в foreground
                StartForeground(NOTIFICATION_ID, CreateNotification());
                
                // инициализируем discord rpc
                discordRPCManager.Initialize("1234567890123456789"); // твой discord app id
                discordRPCManager.UpdatePresence(status, details, state);
                isRunning = true;
            }
            
            return StartCommandResult.Sticky;
        }

        public override IBinder OnBind(Intent intent)
        {
            return null;
        }

        public override void OnDestroy()
        {
            base.OnDestroy();
            if (discordRPCManager != null && isRunning)
            {
                discordRPCManager.Shutdown();
                isRunning = false;
            }
        }

        // создаем канал уведомлений для android 8+
        private void CreateNotificationChannel()
        {
            if (Build.VERSION.SdkInt >= BuildVersionCodes.O)
            {
                NotificationChannel serviceChannel = new NotificationChannel(
                    CHANNEL_ID,
                    "Discord RPC Service",
                    NotificationImportance.Default
                );
                serviceChannel.Description = "управляет discord rich presence";
                
                NotificationManager manager = (NotificationManager)GetSystemService(NotificationService);
                manager.CreateNotificationChannel(serviceChannel);
            }
        }

        // создаем уведомление для foreground сервиса
        private Notification CreateNotification()
        {
            NotificationCompat.Builder builder = new NotificationCompat.Builder(this, CHANNEL_ID)
                .SetContentTitle("Storm Discord RPC")
                .SetContentText("discord rich presence активен")
                .SetSmallIcon(Resource.Drawable.ic_notification)
                .SetOngoing(true)
                .SetColor(Color.ParseColor("#533483")); // фиолетовый цвет

            return builder.Build();
        }
    }
}

File: Android/src/main/java/com/storm/discordrpc/DiscordRPCService.java
package com.storm.discordrpc;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.Service;
import android.content.Intent;
import android.os.Build;
import android.os.IBinder;
import androidx.core.app.NotificationCompat;

// фоновый сервис для работы с discord rpc
public class DiscordRPCService extends Service {
    private static final String CHANNEL_ID = "DiscordRPCChannel";
    private static final int NOTIFICATION_ID = 1;
    
    private DiscordRPCManager discordRPCManager;
    private boolean isRunning = false;

    @Override
    public void onCreate() {
        super.onCreate();
        createNotificationChannel();
        discordRPCManager = new DiscordRPCManager();
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        if (intent != null && !isRunning) {
            // получаем данные из интента
            String status = intent.getStringExtra("status");
            String details = intent.getStringExtra("details");
            String state = intent.getStringExtra("state");

            // запускаем сервис в foreground
            startForeground(NOTIFICATION_ID, createNotification());
            
            // инициализируем discord rpc
            discordRPCManager.initialize("1234567890123456789"); // твой discord app id
            discordRPCManager.updatePresence(status, details, state);
            isRunning = true;
        }
        
        return START_STICKY;
    }

    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (discordRPCManager != null && isRunning) {
            discordRPCManager.shutdown();
            isRunning = false;
        }
    }

    // создаем канал уведомлений для android 8+
    private void createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel serviceChannel = new NotificationChannel(
                    CHANNEL_ID,
                    "Discord RPC Service",
                    NotificationManager.IMPORTANCE_DEFAULT
            );
            serviceChannel.setDescription("управляет discord rich presence");
            
            NotificationManager manager = getSystemService(NotificationManager.class);
            manager.createNotificationChannel(serviceChannel);
        }
    }

    // создаем уведомление для foreground сервиса
    private Notification createNotification() {
        return new NotificationCompat.Builder(this, CHANNEL_ID)
                .setContentTitle("Storm Discord RPC")
                .setContentText("discord rich presence активен")
                .setSmallIcon(R.drawable.ic_notification)
                .setOngoing(true)
                .setColor(0xFF533483) // фиолетовый цвет
                .build();
    }
}
